<?xml version="1.0"?>
<launch>

<!-- General arguments -->
  <arg name="base_name"         default="movo_camera"/>
  <arg name="nodelet_manager"   default="$(arg base_name)"/>
  <arg name="queue_size"        default="5"/>
  <arg name="respawn"           default="true"/>
  <arg name="remap_qhd_to_point_cloud"       default="true"/>
  <arg name="point_cloud_ns"    default="point_cloud"/>
  <arg name="points_topic"      default="points"/>

  <!-- qhd point cloud (960 x 540) -->
  <node pkg="nodelet" type="nodelet" name="$(arg base_name)_points_xyzrgb_qhd" 
        args="load depth_image_proc/point_cloud_xyzrgb $(arg nodelet_manager)" respawn="$(arg respawn)" if="$(arg remap_qhd_to_point_cloud)">
    <remap from="depth_registered/points"     to="$(arg base_name)/$(arg point_cloud_ns)/$(arg points_topic)"/>
    <param name="queue_size" type="int" value="$(arg queue_size)"/>
  </node>


  <node pkg="nodelet" type="nodelet" name="$(arg base_name)_points_xyzrgb_sd"
        args="load depth_image_proc/point_cloud_xyzrgb $(arg nodelet_manager)" respawn="$(arg respawn)">
    <remap from="rgb/camera_info"             to="$(arg base_name)/sd/camera_info"/>
    <remap from="rgb/image_rect_color"        to="$(arg base_name)/sd/image_color_rect"/>
    <remap from="depth_registered/image_rect" to="$(arg base_name)/sd/image_depth_rect"/>
    <remap from="depth_registered/points"     to="$(arg base_name)/sd/points"/>
    <param name="queue_size" type="int" value="$(arg queue_size)"/>
  </node>

</launch>
